pkg/parser/parser.go

```diff
@@ -518,18 +527,28 @@ func (p *Parser) parseIfBlock(scope *scope) *BlockStatement {
 }
 
 func (p *Parser) parseBlockWithEndTokens(scope *scope, endTokens map[lexer.TokenType]bool) *BlockStatement {
-	tok := p.cur
-	var stmts []Node
+	block := &BlockStatement{Token: p.cur}
+	var prev Node
 	for !endTokens[p.cur.TokenType()] {
+		tok := p.cur
 		stmt := p.parseStatement(scope)
-		if stmt != nil {
-			stmts = append(stmts, stmt)
+		if stmt == nil {
+			continue
+		}
+		if alwaysReturns(prev) {
```

Comment:
We don't need `prev` any more. This can be `if block.AlwaysReturns()` (or `if block.alwaysReturns`) since we set that on the previous iteration.

Reply:
yes, nice! I will make alwaysReturns private....